src/golang/cmd/server/server/refresh_workflow.go

```diff
@@ -108,5 +109,13 @@ func (h *RefreshWorkflowHandler) Perform(ctx context.Context, interfaceArgs inte
 	if err != nil {
 		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to trigger this workflow.")
 	}
+
+	// TODO: debate this in the PR
+	status, err := job.PollJob(ctx, jobName, h.JobManager, 2*time.Second, 5*time.Minute)
+	if err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unable to poll this workflow.")
+	}
+	fmt.Printf("Cron job completed with status %s\n", status)
+
```

Comment:
I don't think polling on the job here is the long term solution. refresh_workflow is called by register_workflow, and it seems silly to block `publish_flow()` until an entire flow run completes.

Any ideas? What if we had a long-running goroutine on every process manager that listens for certain jobs to complete, and flushes the stdout and stderr of those jobs to the console + deletes the cmd? We wouldn't have to poll() after every launch() to get the logs then, we could just launch() and then register the jobname with this background thread.

Reply:
Actually yeah I think you're right! I guess I didn't retest correctly after making all the changes.